ouroboros-consensus/bench/ChainSync-client-bench/Main.hs

```diff
@@ -0,0 +1,302 @@
+
+securityParam :: SecurityParam
+securityParam = SecurityParam $ fromIntegral kInt
+
+initialChain :: NE.NonEmpty B
+initialChain =
+    NE.fromList
+  $ take kInt
+  $ iterate TB.successorBlock
+  $ TB.firstBlock 0
+
+-----
+
+slotLengthInSeconds :: Int
+slotLengthInSeconds = 1
+
+slotLength :: SlotLength
+slotLength = slotLengthFromSec $ toEnum slotLengthInSeconds
+
+numCoreNodes :: NumCoreNodes
+numCoreNodes = NumCoreNodes 2
+
+topConfig :: TopLevelConfig B
+topConfig = TopLevelConfig {
+    topLevelConfigProtocol = BftConfig {
+        bftParams  = BftParams {
+                         bftSecurityParam = securityParam
+                       , bftNumNodes      = numCoreNodes
+                       }
+      , bftSignKey = SignKeyMockDSIGN 0
+      , bftVerKeys = Map.fromList [
+                         (CoreId (CoreNodeId 0), VerKeyMockDSIGN 0)
+                       , (CoreId (CoreNodeId 1), VerKeyMockDSIGN 1)
+                       ]
+      }
+  , topLevelConfigLedger  = eraParams
+  , topLevelConfigBlock   = TB.TestBlockConfig numCoreNodes
+  , topLevelConfigCodec   = TB.TestBlockCodecConfig
+  , topLevelConfigStorage = TB.TestBlockStorageConfig
+  }
+  where
+    eraParams :: HardFork.EraParams
+    eraParams = HardFork.defaultEraParams securityParam slotLength
+
+-----
+
+data FollowerState =
+    Resting !(RealPoint B)
+  |
+    Switching !(Point B) !(NE.NonEmpty B) 
+  |
+    Switched !(NE.NonEmpty B) 
+
+-- | A 'ChaindB.Follower' that merely switches between each of the
+-- `kInt`-length chains, in the order enumerated by 'TB.updateToNextNumeral'.
+--
+-- INVARIANT: the chains are never longer than 'kInt' and are 'kInt' long
+-- infinitely often.
+mkFollower ::
+     StrictTVar IO (Tip B)
+  -> IO (ChainDB.Follower IO B (ChainDB.WithPoint B H))
+mkFollower varTip = do
+    varState <- newIORef $ Resting $ blockRealPoint $ NE.last initialChain
+
+    let wrap blk = ChainDB.WithPoint (getHeader blk) (blockPoint blk)
+
+    let next = readIORef varState >>= \case
+            Switching ipoint blks -> do
+                writeIORef varState $ Switched blks
+                atomically $ writeTVar varTip $ tipFromHeader $ NE.last blks
+                pure $ RollBack ipoint
+            Switched blks -> do
+                let blk = NE.head blks
+                writeIORef varState $ case NE.nonEmpty (NE.tail blks) of
+                    Nothing    -> Resting $ blockRealPoint blk
+                    Just blks' -> Switched blks'
+                atomically $ writeTVar varTip $ tipFromHeader $ NE.last blks
+                pure $ AddBlock $ wrap blk
+            Resting rp -> do
+                let (ipoint, blks) = TB.updateToNextNumeral rp
+                writeIORef varState $ Switched blks
+                atomically $ writeTVar varTip $ tipFromHeader $ NE.last blks
+                pure $ RollBack ipoint
+
+    pure ChainDB.Follower {
+        ChainDB.followerClose = pure ()
+      ,
+        ChainDB.followerForward = \case
```

Comment:
```suggestion
        ChainDB.followerClose = pure ()
      
      , ChainDB.followerForward = \case
```

I would prefer commas before names, even if we leave an empty line in between. But this is just an opinion.

Reply:
Done